### ISSUE TITLE
Add Support of Spring Data JPA specific JpaSpecificationExecutor

### ISSUE THREAD

**Issue by hantsy** [quarkusio/quarkus#4040]
Quarkus 0.22.0

I just tried Spring Data JPA feature, the first limitation is JpaSpecificationExecutor does not work.

https://github.com/hantsy/quarkus-sample/tree/master/spring-post-service

```java
public interface PostRepository extends JpaRepository<Post, String>//, JpaSpecificationExecutor<Post>
{

}
```

**Comment by geoand** [comment]
We have made it  clear that not everything from Spring Data will work on Quarkus.
Can you give some details about how you usually use `JpaSpecificationExecutor`?

**Comment by hantsy** [comment]
Some like the purpose I described in issue #4016 and #3965.

 `JpaSpecificationExecutor` is to support JPA Criteria and also introduces a great `Specification` (a DDD concept) to satisfy **Type-safe** query requirements.

Many developers are practicing **Typesafe** query instead of string based query in development.

**Comment by geoand** [comment]
Thanks for the info.

When we have a general strategy around how Panache will handle such `Typesafe` handling, we can revisit this

**Comment by hantsy** [comment]
@geoand Micronaut Data ports this `Specification` concept(a **type safe** approach to compose the query criteria) to almost all data modules, including Data Jdbc, Data R2dbc, and Data Hibernate Reactive.

**Comment by geoand** [comment]
@Michael-AT-Corporation correct. We are likely to figure out the type-safe query strategy soon (cc @FroMage).
In the meantime,  I opened #42216 to update the docs

**Comment by Michael-AT-Corporation** [comment]
@geoand Thanks for the info and for updating the documentation 🙂

**Comment by geoand** [comment]
🙏🏼

**Comment by FroMage** [comment]
We are adding type-safe queries, but using HQL, not specifications. However thanks for the point, I've noted the use-case.

**Comment by pipinet** [comment]
@geoand @FroMage @hantsy 
Is there support for a JpaSpecificExecutor program? Because currently he can easily support [Hibernate Data Repositories](https://docs.jboss.org/hibernate/orm/6.6/repositories/html_single/Hibernate_Data_Repositories.html). 
[Micronaut Data](https://micronaut-projects.github.io/micronaut-data/latest/guide/#repositories) also provides similar support.


This allows us to write more convenient、dynamic and typed JPQL。
